packages/@aws-cdk/aws-apigatewayv2-authorizers/lib/http/lambda.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  HttpAuthorizer,
+  HttpAuthorizerType,
+  HttpRouteAuthorizerType,
+  HttpRouteAuthorizerBindOptions,
+  HttpRouteAuthorizerConfig,
+  IHttpRouteAuthorizer,
+  AuthorizerPayloadFormatVersion,
+} from '@aws-cdk/aws-apigatewayv2';
+import { IFunction } from '@aws-cdk/aws-lambda';
+import { Token, Stack, Duration } from '@aws-cdk/core';
+
+/**
+ * Specifies the type of lambda authorizer
+ */
+export enum HttpLambdaAuthorizerType {
+  /** Returns simple boolean response */
+  SIMPLE,
+
+  /** Returns an IAM Policy */
+  AWS_IAM,
+}
+
+/**
+ * Properties to initialize HttpTokenAuthorizer.
+ */
+export interface HttpLambdaAuthorizerProps {
+
+  /**
+   * The name of the authorizer
+   * @default 'LambdaAuthorizer'
+   */
+  readonly authorizerName?: string;
+
+  /**
+   * The identity source for which authorization is requested.
+   *
+   * @default ['$request.header.Authorization']
+   */
+  readonly identitySource?: string[];
+
+  /**
+   * Specifies the format of the payload sent to an HTTP API Lambda authorizer.
+   *
+   * @default - 2.0
+   */
+  readonly payloadFormatVersion?: AuthorizerPayloadFormatVersion;
```

Comment:
As far as I understand, the difference between 1.0 and 2.0  is that 2.0 allows both IAM and SIMPLE and 1.0 only allows SIMPLE. Any other differences?

If that's the case, we can keep this simple. 

```ts
new HttpLambdaAuthorizer({
  ...
  responseTypes: [ HttpLambdaResponseType.IAM, HttpLambdaResponseType.SIMPLE ]
});
```

We can manipulate the `EnableSimpleResponses` and `AuthorizerPayloadFormatVersion` based on the values provided.

Reply:
Good idea ✨

Reply:
Actually now that I think about this, what about persons that want to return IAM but use version 2? Think we should allow this to stay configurable

Reply:
Think a good middle ground is to remove `payloadFormatVersion`, and derive it from the type.

Reply:
Looks like there's no combination to specify that you want "only" simple. So maybe like this?

| responseTypes | AuthorizerPayloadFormatVersion | EnableSimpleResponses |
| ------ | ----- | ----- |
| IAM | 2.0 | false |
| IAM_SIMPLE | 2.0 | true

Is there any use case to use 1.0 instead of 2.0?

Reply:
> If you need compatibility with REST APIs, use version 1.0.

All I saw. 

Will make the change though